### ISSUE TITLE
Upgrade Page Not Recognizing Click from Formtools Upgrade Link

### ISSUE THREAD

**Issue by john-m-adams** [formtools/core#638]
**tldr; Redirecting from http to https breaks POST.**

Hello,

Current instance is on Formtools 2.2.6

When I click "Update", it opens the upgrade page, however it gives this message:

> Upgrade
> This page cannot be called directly. You must click the "Update" link in your Form Tools installation.
> 
> Form Tools Upgrade Script 2.0.1

I attempted this with the same results from both my "www" and "test" hosts/installations.

I also attempted this from several different pages:
/formtools/admin/forms/
/formtools/admin/clients/
/formtools/admin/modules/
etc...

I also notice it posts to:
http://www.formtools.org/upgrade.php
which is rewritten to:
https://formtools.org/upgrade/

I attempted to use the inspector to change the form's post address to https://formtools.org/upgrade/, but then clicking the link yielded an HTTP Error 500.
(Note, I think almost any form that's HTTPS gets that same result... it's only switching protocol which breaks the POST request).

I imagine the first error is POST switching to GET, with change in protocol (and host?), and then the HTTP Error 500 on changing the page in the inspector is actually a success, but reaches the standard server error reported elsewhere (Most recently here #635 ).

Thank you!
John

**Comment by benkeen** [comment]
Hey John, 

Sorry for the wait - heck of a week and I'm going away for 10 days on Friday. Ack! 

I checked this out and yes, I can confirm it. I'd added in some code to store the data before it switches patches + protocols but possibly I broke it on a recent update. I'll keep you posted.

**Comment by john-m-adams** [comment]
Thank you!

If it's faster to fix the Error 500 first, the redirect is easy to fix as a user, by changing the url in the browser's inspector.  I can post instructions for other users, if the 500 error is fixed before the redirect.

**Comment by john-m-adams** [comment]
Workaround:

If your upgrade link reaches this error, open your browser's inspector F12, or Control-Shift-I, or Command-Shift-I, in most browsers.

Find this element:
`<form action="http://www.formtools.org/upgrade.php" id="upgrade_form" method="post" target="_blank">`

Use "Edit as HTML" or "Edit Attribute" to change `http://www.formtools.org/upgrade.php` to `https://www.formtools.org/upgrade/`

Click the "Upgrade" link again, and it should work.
